src/withElementPortal.js

```diff
@@ -5,21 +5,18 @@ function getDisplayName(Component) {
   return Component.displayName || Component.name || 'Component';
 }
 
-const withElementPortal = (Child) => {
-  const Portal = ({ id, selector, shouldReset, mapDomNodeToProps, ...childProps }) => {
-    const ChildWrapper = (props) => <Child {...childProps} {...props} />;
-    return <ElementPortal
-      id={id}
-      selector={selector}
-      mapDomNodeToProps={mapDomNodeToProps}
-      shouldReset={shouldReset}
-      view={ChildWrapper}
-    />;
-  };
+const withElementPortal = (portalProps) => (component) => {
+  const WithElementPortal = (props) => (
+    <ElementPortal
+      {...portalProps}
+      component={(mappedProps) => (
+        React.createElement(component, {...props, ...mappedProps})
```

Comment:
minor: shouldn't this be `Component` (capitalized)?

Reply:
@vitorbal I'm going to change it back to `Component`, you're right, that's a constructor, it should be capitalized.